theopensystemslab/planx-new
# feat: setup invitation, reminder, and expiry emails for payment requests using Hasura event triggers

when a `payment_requests` record is inserted (do each x2 for payee & agent):
- immediately send an invitation to pay email
- schedule a reminder email in 28-7 days
- schedule an expiry email in 28 days

todo: 
- [x] update s3 secrets with new Notify template ids
- [x] populate basic email body templates with session preview data, team notify personalisation, etc - share with Ian/design team for feedback
- [ ] clean up scheduled reminders & expiry events when the `payment_request` is marked as paid - would vote this is a done in a follow-up PR! 

to test: 
- generate a lowcal_session record, copy the sessionId and manually insert a related `payment_requests` record via the Hasura console (or "clone" an existing row, swapping in your own email)
- click "save" to insert: you should get 2 emails in your inbox (assumes you've used your email for both the session & payment)
- you should be able to check the Hasura "events" tab and see successfully scheduled events for reminder & expiry emails
